Packages/DesignSystem/Sources/DesignSystem/ThemeApplier.swift

```diff
@@ -58,6 +62,26 @@ struct ThemeApplier: ViewModifier {
   }
 
   #if canImport(UIKit)
+    private func setWindowUserInterfaceStyle(from colorScheme: ColorScheme) {
+      guard !theme.followSystemColorScheme else {
+          setWindowUserInterfaceStyle(.unspecified)
+          return
```

Comment:
That's fixing the problem in my comment right?

Reply:
Yes, this undoes the overriding of the user interface style by resetting it to `.unspecified` in case the system theme is followed

A recording of the issue when not following system mode without this PR.

iPhone 14, iOS 16.2 - Light theme for system

| Settings | Issue |
|:-:|:-:|
| ![Settings](https://user-images.githubusercontent.com/6899256/215322188-086867b8-3777-4730-86c1-2b6eeeb927d4.png) | https://user-images.githubusercontent.com/6899256/215322215-99f69d27-21eb-4d45-83c2-399c55fea5fa.mov |